### DISCUSSION TITLE
How to disable the Viser viewer during training Nerfacto?

### DISCUSSION THREAD

**Discussion by AruniRC** [nerfstudio-project/nerfstudio#3258] [Q&A]
I am running the Nerfacto training from Python (code snippet shown below). Everything works perfectly, except I cannot turn off the Viser viewer automatically starting. Is there any way to programmatically turn off Viser during training? 

If this cannot be done currently, I think making a feature request for this makes sense. In some applications, building something on top of Nerfstudio, we may not want to use Viser but some custom tool for the viewer. Being able to easily turn off Viser via the Python API at least would help here.

```
from nerfstudio.configs.method_configs import method_configs
from nerfstudio.scripts.train import main
from pathlib import Path

# Create default training config for the nerfacto method
cfg = method_configs["nerfacto"]

# Specify where the pre-processed data is saved (must contain 'transforms.json')
cfg.data = Path('processed_data_folder')

# turn off in-browser viewer (Viser) ?
# cfg.vis = None        # no effect!
# cfg.viewer = None  # training fails with error!

# begin training!
main(cfg)

```

**Comment by nomppy** [comment]
any update on this? it's weird that we are not given to turn off this feature, given that the viewer tends to run indefinitely even after training is done!

**Reply by AruniRC** [reply]
Changing the viewer to tensorboard or wandb without viser is a hacky workaround that I ended up using.
